### DISCUSSION TITLE
Telling Textual App how to ignore or pass-through certain keystroke sequences to originating stdin

### DISCUSSION THREAD

**Discussion by evanpolster** [Textualize/textual#2950] [Q&A] [answered]
Use Case: We have a requirement that when we run long-running jobs from a terminal (like a lengthy build) we must do it from inside a `tmux` session. That is because our security configuration sets an user-input inactivity timeout on our connection and kicks us out before the job has finished. The `tmux` session allows us to reconnect to the system and reattach to the terminal session (i.e. the still running build) as if we were never detached to begin with.

I tried this morning to run my Textual app (which indeed wraps our long-running build) inside a `tmux` session and it appears to work correctly which is potentially very powerful. 

The only thing I can't figure out how to do is send the `tmux` magical 'detach' keystroke sequence (e.g. `Ctrl-B D`) from inside the Textual app in order to get out of the tmux session and back up into the original default terminal (i.e. from where I ran `tmux new-session` to begin with). 

As Textual appears to do with `Ctrl-C`, is there any way to tell the Textual app to pass through certain key sequences from its intercepted stdin back to the command-line's stdin (i.e. the terminal session from where the Textual app was started)?  

If not, is there a key sequence that can push the Textual App  to the background (kinda like an equivalent to Ctrl-Z on Linux shell )?

**Comment by davep** [comment, marked as answer]
To clarify on the <kbd>Ctrl</kbd>+<kbd>C</kbd> front: that's something that's [handled by Textual itself](https://github.com/Textualize/textual/blob/ec58b914cf902bd6056b6c285a2bb9ade02a3537/src/textual/app.py#L287-L289) -- it's not that the key combination gets passed "up" and the operating system kills the application.

On your more general point, just to check I'm understanding things correctly: you want some way to start a tmux session, running a Textual app, and have it detach right away so you can come back to it later? (I ask only because <kbd>Ctrl</kbd>+<kbd>B</kbd> <kbd>D</kbd> works right now with Textual apps, if you're the one pressing it). If that's the case, would it not make more sense to create a detached session to start with?

I'm not really much of a tmux user but I believe this should work just fine:

```sh
$ tmux new-session -d my-textual-application-here
```

then later you can:

```sh
$ tmux attach
```

to attach to the session and view and interact with the application.

Testing that here now and it all works fine.

**Reply by evanpolster** [reply]
You are correct. I apparently had my tmux detach key sequence set to `Ctrl-a d` and not the default of `Ctrl-b d` 

Thanks, as usual, for your excellent answer and apologies for the disruption.

**Reply by davep** [reply]
Not at all disruptive, glad you got the answer one way or another. 🙂
